### PR TITLE
Workflow Invocation Methods

### PR DESCRIPTION
**Pull request by kraftp**
Documentation for https://github.com/dbos-inc/dbos-transact/pull/432

### CODE REVIEW

File: docs/api-reference/contexts.md [hunk 1]
```diff
@@ -158,20 +158,49 @@ The [Koa Context](https://github.com/koajs/koa/blob/master/docs/api/context.md)
...
 ```typescript
-handlerCtxt.invoke(Bar).foo(baz)
+const output = await handlerCtxt.invoke(Cls).fn(arg)
 ```
 
-When calling transactions or communicators, `invoke()` asynchronously returns the function's output.
-When calling workflows, `invoke()` asynchronously returns a [`handle`](./workflow-handles) for the workflow.
-
-Note that the DBOS runtime will supply the context to invoked functions.
+You don't need to supply a context to the invoked function&#8212;the DBOS Transact runtime does this for you.
```
Comment (chuck-dbos) [inline docs/api-reference/contexts.md:169]: Unicode char necessary?

File: docs/api-reference/contexts.md [hunk 2]
```diff
@@ -158,20 +158,49 @@ The [Koa Context](https://github.com/koajs/koa/blob/master/docs/api/context.md)
...
 You can optionally provide a UUID idempotency key to the invoked function.
 For more information, see our [idempotency tutorial](../tutorials/idempotency-tutorial.md).
 
+#### `handlerCtxt.invokeWorkflow`
+
+```typescript
+invokeWorkflow<T>(targetClass: T, workflowUUID?: string): InvokeFuncs<T>
+```
+
+Invoke a workflow and wait for it to complete, returning its result.
```
Comment (chuck-dbos) [inline docs/api-reference/contexts.md:179]: I know it's right below here but I'd mention start.  (If you do not want to wait for the workflow to complete, see startWorkflow below.)

File: docs/api-reference/contexts.md [hunk 3]
```diff
@@ -225,31 +255,47 @@ invoke<T>(targetClass: T, workflowUUID?: string): InvokeFuncs<T>
...
+invokeChildWorkflow<T extends any[], R>(wf: Workflow<T, R>, ...args: T): Promise<R>
+```
 
-#### `workflowCtxt.childWorkflow`
+Invoke a child workflow and wait for it to complete, returning its result.
+The syntax for invoking workflow `wf` in class `Cls` with argument `arg` is:
 
 ```typescript
-childWorkflow<T extends any[], R>(wf: Workflow<T, R>, ...args: T): Promise<WorkflowHandle<R>>
+const output = await ctxt.invokeChildWorkflow(Cls.wf, arg)
```
Comment (chuck-dbos) [inline docs/api-reference/contexts.md:278]: ;

File: docs/api-reference/contexts.md [hunk 4]
```diff
@@ -225,31 +255,47 @@ invoke<T>(targetClass: T, workflowUUID?: string): InvokeFuncs<T>
...
-The syntax for invoking workflow `foo` in class `Bar` with argument `baz` is:
+#### `workflowCtxt.startChildWorkflow`
 
 ```typescript
-const workflowHandle = await ctxt.childWorkflow(Bar.foo, baz)
+startChildWorkflow<T extends any[], R>(wf: Workflow<T, R>, ...args: T): Promise<WorkflowHandle<R>>
 ```
 
+Durably start a child workflow and return a [handle](./workflow-handles.md) to it but do not wait for it to complete.
+This method resolves as soon as the handle is safely created; at this point the workflow is guaranteed to [run to completion](../tutorials/workflow-tutorial.md#reliability-guarantees).
```
Comment (chuck-dbos) [inline docs/api-reference/contexts.md:290]: safely -> durably?  And maybe remove "Durably" above, as there's no other way we do it.

File: docs/tutorials/workflow-tutorial.md [hunk 5]
```diff
@@ -45,9 +45,14 @@ For example, this line from our above example invokes the transaction `InsertGre
...
 
-The syntax for invoking function `foo(args)` in class `Bar` is `ctxt.invoke(Bar).foo(args)`.
+The syntax for invoking function `fn(args)` in class `Cls` is `ctxt.invoke(Cls).fn(args)`.
 
-You can also invoke other workflows using the [ctxt.childWorkflow()](../api-reference/contexts#workflowctxtchildworkflow) method.
+You can also invoke other workflows with the [`ctxt.invokeChildWorkflow()`](../api-reference/contexts#workflowctxtinvokechildworkflow) method.
+The syntax for invoking child workflow `wf` in class `Cls` with argument `arg` is:
+
+```typescript
+const output = await ctxt.invokeChildWorkflow(Cls.wf, arg)
```
Comment (chuck-dbos) [inline docs/tutorials/workflow-tutorial.md:54]: Why is it not invokeChildWorkflow(Cls).wf(arg) like the other invokes?

File: docs/tutorials/workflow-tutorial.md [hunk 6]
```diff
@@ -97,19 +105,13 @@ You can also retrieve another workflow's handle if you know its identity:
   }
 ```
 
-To wait for a workflow to complete and retrieve its result, await `handle.getResult()`:
+To wait for a workflow to complete and retrieve its result, use `handle.getResult()`:
```
Comment (chuck-dbos) [inline docs/tutorials/workflow-tutorial.md:108]: No need to await?

File: docs/tutorials/workflow-tutorial.md (hunk 5 above)
Comment (kraftp) [reply docs/tutorials/workflow-tutorial.md]: Discussed in https://github.com/dbos-inc/dbos-transact/pull/432

**Review by qianl15 — APPROVED** [review]

**Pull request merged**